tests/shopify_python/test_google_styleguide.py

```diff
@@ -225,3 +225,57 @@ def test_use_cond_exprs(self):
         # root_not_msg should add no messages
         with self.assertNoMessages():
             self.walk(root_not_msg)
+
+    def test_unary_lambda_func(self):
+        unary_root = astroid.builder.parse("""
+        def unaryfnc():
+            unary_pass = map(lambda x: not (x+3), [1, 2, 3, 4])
+            unary_msg = map(lambda x: -x, [1, 2, 3, 4])
+        """)
+
+        # Unary Op Test 1: This will not trigger a message
+        ulam_pass = unary_root.body[0].body[0].value.args[0]
+        with self.assertNoMessages():
+            self.walk(ulam_pass)
+
+        # Unary Op Test 2: This will trigger a message
+        ulam_msg = unary_root.body[0].body[1].value.args[0]
+        unary_message = pylint.testutils.Message('lambda-func', node=ulam_msg, args={
+            'op': 'operator.neg',
+            'lamfun': 'lambda x: - x'
+            })
+        with self.assertAddsMessages(unary_message):
+            self.walk(ulam_msg)
+
+    def test_binary_lambda_func(self):
+        binary_root = astroid.builder.parse("""
+        def binaryfnc():
+            binary_pass = reduce(lambda x, y, z: x * y + z, [1, 2, 3])
+            binary_pass2 = map(lambda x: x + 3, [1, 2, 3, 4])
+            binary_pass3 = map(lambda x, y: x + 1)
+            binary_msg = reduce(lambda x, y: x * y, [1, 2, 3, 4])
+        """)
+
+        # Binary Op Test 1: This will not trigger a message
```

Comment:
Alternatively, it could be one test, but you pass the whole thing in at once and assert that there is only one failure. Could be made a bit more clear by naming it `binary_fail`